lib/ui/page/buy.dart

```diff
@@ -0,0 +1,437 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+import 'package:flutter_hooks/flutter_hooks.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+import '../../db/mixin_database.dart';
+import '../../service/env.dart';
+import '../../util/constants.dart';
+import '../../util/debouncer.dart';
+import '../../util/extension/extension.dart';
+import '../../util/hook.dart';
+import '../../util/l10n.dart';
+import '../../util/logger.dart';
+import '../../util/r.dart';
+import '../../wyre/wyre_client.dart';
+import '../../wyre/wyre_constants.dart';
+import '../../wyre/wyre_quote.dart';
+import '../../wyre/wyre_vo.dart';
+import '../router/mixin_routes.dart';
+import '../widget/asset_selection_list_widget.dart';
+import '../widget/brightness_observer.dart';
+import '../widget/buttons.dart';
+import '../widget/fiat_selection_list_widget.dart';
+import '../widget/mixin_appbar.dart';
+import '../widget/mixin_bottom_sheet.dart';
+import '../widget/round_container.dart';
+import '../widget/symbol.dart';
+
+class Buy extends HookWidget {
+  const Buy({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    var assetId = context.pathParameters['id'];
+    assetId ??= erc20USDT;
+
+    final supportedAssets = useMemoizedFuture(
+        () => context.appServices.findOrSyncAssets(supportedCryptosId)).data;
+    if (supportedAssets == null || supportedAssets.isEmpty) {
+      w('supportedAssets: $supportedAssets');
+      return const SizedBox();
+    }
+
+    final fiatList = useMemoized<List<WyreFiat>>(getWyreFiatList);
+    return _Buy(supportedAssets: supportedAssets, fiatList: fiatList);
+  }
+}
+
+class _Buy extends HookWidget {
+  _Buy({
+    Key? key,
+    required this.supportedAssets,
+    required this.fiatList,
+  }) : super(key: key);
+
+  final List<AssetResult> supportedAssets;
+  final List<WyreFiat> fiatList;
+
+  final _debouncer = Debouncer(const Duration(milliseconds: 500));
```

Comment:
init `debouncer` with `userMemo()` in `build(BuildContext)` method might be better?

Reply:
implementation changed